### ISSUE TITLE
quote format problem

### ISSUE THREAD

**Issue by karisawa** [9fans/plan9port#87]
I have been wondering why some Japanese rune are quoted which, I think, should not be quoted.
The example is "未". The UTF-8 expression is (e6 9c aa).
To day, I found suspicious code.

$PLAN9/src/lib9/needsrcquote.c
int
needsrcquote(int c)
{
	if(c <= ' ')
		return 1;
	if(strchr("`^#*[]=|\\?${}()'<>&;", c))
		return 1;
	return 0;
}

the code below will fix the problem.

int
needsrcquote(int c)
{
	if(c & ~0x7f)
		return 0;
	if(c <= ' ' || strchr("`^#*[]=|\\?${}()'<>&;", c))
		return 1;
	return 0;
}

**Comment by rsc** [comment]
Thanks. That's embarrassing.

**Comment by vat9** [comment]
One day I’ll figure out MailScanner’s tuning options to trim and remove,
until then at least the GitHub javascript goo isn’t pushed through my mailers.

HTML email has to be one of the worst inventions, maybe even as bad as XML.
Though if I were tracking details on processed email on remote machines I’d
be of a different opinion, HTML is fantastic for that type of utilization.

**Comment by rsc** [comment]
You can't tell since it's being removed, but there's no Javascript in the mail from Github. It's a JSON blob giving metadata about the email. It actually looks completely reasonable:

```
<script type="application/json" data-scope="inboxmarkup">
{
	"api_version": "1.0",
	"publisher": {
		"api_key": "05dde50f1d1a384dd78767c55493e4bb",
		"name": "GitHub"
	},
	"entity": {
		"external_key": "github/9fans/plan9port",
		"title": "9fans/plan9port",
		"subtitle": "GitHub repository",
		"main_image_url": "https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png",
		"avatar_image_url": "https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png",
		"action": {
			"name": "Open in GitHub",
			"url": "https://github.com/9fans/plan9port"
		}
	},
	"updates": {
		"snippets": [
			{
				"icon": "DESCRIPTION",
				"message": "Closed #87 via 1f1ab4ccbbec9c92b780f6a60ff9730126659a87."
			}
		],
		"action": {
			"name": "View Issue",
			"url": "https://github.com/9fans/plan9port/issues/87#event-1129146157"
		}
	}
}
</script>
```
